G/GDAL/build_tarballs.jl

```diff
@@ -0,0 +1,74 @@
+using BinaryBuilder
+
+name = "GDAL"
+version = v"3.0.2"
+
+# Collection of sources required to build GDAL
+sources = [
+    "https://github.com/OSGeo/gdal/releases/download/v$version/gdal-$version.tar.gz" =>
+    "787cf150346e58bff0ccf8c131b333139273e35d2abd590ad7196a9ee08f0039",
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/gdal-*/
+
+if [[ ${target} == *mingw* ]]; then
+    export LDFLAGS="-L${libdir}"
+    cp ${libdir}/libproj_6_2.dll ${libdir}/libproj.dll
```

Comment:
`libgdal` needs `libproj`. However, due to `PROJ_jll` with it's `libproj_6_2.dll` being a dependency of `GDAL_jll`, won't `libproj_6_2.dll` already be `dlopen`ed before `libgdal` and thus everything is fine?

Reply:
That's not a matter of being dlopened, when the linker is passed the option `-lfoo` it'll look for a file called `libfoo.${dlext}` (I think on Windows the "lib" is not necessary), with some variations to allow for the version number.  Here the file is called `libproj_6_2.dll`, the linker can't find `libproj.dll` (probably it would allow for `libproj-62.dll`).

Reply:
Interesting. But in [GDALBuilder](https://github.com/JuliaGeo/GDALBuilder/blob/afb7e5fbeb879fd8829a295df31b44df263db865/build_tarballs.jl#L18-L20) we used the same trick and dit not distribute `libproj.dll`, only `libproj_6_1.dll`. And it does find it:
![image](https://user-images.githubusercontent.com/4471859/71037039-354a8380-211f-11ea-80b0-ea3518ec5c81.png)

Reply:
As far as I can see you're not deleting the file at the end, so it's probably in GDAL tarball, which is definitely something that we don't want.  As I said before, the linker should allow for the version of the package to be in the file name, but I've never seen it as `libproj_6_1`, I'm surprised the loader can find it

Reply:
Hmm in the old `GDAL/lib` there is `libproj.dll.a`, but that's all, and if I move that file away it still finds it just fine.

So at least we should add a `rm ${libdir}/libproj.dll` after compilation in this build script to prevent getting that file out of this build? And if we get issues we try to fix the name in the PROJ build script?

Reply:
The libraries for Windows are under bin, not lib

Reply:
Yeah I know, I just meant that under bin there is only `libproj_6_1.dll`, nothing else. So I gues the loader on Windows is sufficiently flexible, just not during the build, which is why we needed to rename it before the build only.

Reply:
Have a look at https://github.com/JuliaPackaging/Yggdrasil/pull/293/commits/b7eae71ec07e09d9ec47d7870d8a086821543634, if you don't like we can hard-reset it.  However, I think this is a better way to deal with this, instead of copying files around.

In an ideal world, PROJ would provide a pkgconfig file and we would get from there the name of the library, instead of doing this mess.

Reply:
Thanks! Have you considered submitting this patch upstream to https://github.com/OSGeo/gdal/commits/master/gdal/configure.ac?